pandas/core/groupby/generic.py

```diff
@@ -1296,6 +1314,26 @@ class DataFrameGroupBy(NDFrameGroupBy):
     A
     1   1   2  0.590716
     2   3   4  0.704907
+
+    To control the output names with different aggregations per column,
+    pandas supports "keyword aggregation"
+
+    >>> df.groupby("A").agg(
+    ...     b_min=pd.KeywordAgg(column="B", aggfunc="min"),
+    ...     c_sum=pd.KeywordAgg(column="C", aggfunc="sum"))
```

Comment:
Do we want to show the namedtuple version here?

Reply:
I think we should encourage the namedtuple version in the docs. What do you think?

Reply:
Speaking as a user, no I don't think that's the best.

The old way was delightfully concise (dict of dicts) but not very explicity, the new way with keywords is slightly less concise (repetition of the selection column) but not by much and more consistent so it's pretty OK. 

But the named tuple version is very verbose and - - unless my code was designed for other people and need to be self documenting - - in the usual case of using pandas in Notebooks for my own work, once the user understands the concept, using the namedtuple version is just making the code less readable due to the lower ratio of user content (column names and aggfunc names) / boilerplate content (pd.NamedAgg, column=, aggfunc=).

Moreover, letting the simple keyword-tuple version having the most emphasis is more likely to help intermediate and advanced python user think that this system of aggregation can be easily made more dynamic inside code through usage of the **{} construct, where the dict to unpack is constructed by the code, and only is a dict of tuples.

Reply:
@TomAugspurger wrote:
> For the documentation I think it’s important to include the namedtuple. I surveyed a few friends and they were split on whether the keyword was the column selection or the output name. I view it similar to attribute access of column names. In my own throwaway code, I’ll use . to select a column. But in library code I’ll use getitem.

I am not saying not to include namedtuple in the documentation. 

The question was whether it should be encouraged or simply mentioned. I think mention is enough. 

The non-namedtuple version is pretty straightforward to understand once you see one example: for example with `.agg(average_weight=('height', np.mean))` it is clear by reading this that the `mean` of the `height` column would go into the column `average_weight`.